### ISSUE TITLE
Need tests for GH #8369

### ISSUE THREAD

**Issue by khwilliamson** [Perl/perl5#19781]
[This issue](https://github.com/Perl/perl5/issues/8369) has been fixed, but there are no real tests for it.  See also #19636

**Comment by hvds** [comment]
I think this is the core test case, here's the output before the check was restored:
```
% ./miniperl -wle 'print $& if "aaaaab" =~ /(?:a|xx){0,4}?b/'
aaaaab
% 
```

**Comment by jkeenan** [comment]
> I think this is the core test case, here's the output before the check was restored:
> 
> ```
> % ./miniperl -wle 'print $& if "aaaaab" =~ /(?:a|xx){0,4}?b/'
> aaaaab
> % 
> ```

Given that most of the time our test suite is run with perl rather than miniperl, could you work this into a test we can add underneath `t/` somewhere?

**Comment by hvds** [comment]
> Given that most of the time our test suite is run with perl rather than miniperl, could you work this into a test we can add underneath `t/` somewhere?

That'd be `./perl -wle 'print $& if "aaaaab" =~ /(?:a|xx){0,4}?b/'`

miniperl functionality is a strict subset of perl functionality - any time we say "miniperl does this" that's shorthand for "perl does this, and requires only core functionality to do so".

**Comment by khwilliamson** [comment]
Thanks, now fixed by b58cce2db66370e792e026280d09baf9caba65b6
